deps/rabbit/src/rabbit_variable_queue.erl

```diff
@@ -2613,14 +2614,33 @@ maybe_deltas_to_betas(DelsAndAcksFun,
                         ram_bytes            = RamBytes,
                         disk_read_count      = DiskReadCount,
                         delta_transient_bytes = DeltaTransientBytes,
-                        transient_threshold  = TransientThreshold }) ->
+                        transient_threshold  = TransientThreshold,
+                        rates                = #rates{out = RatesOut }}) ->
     #delta { start_seq_id = DeltaSeqId,
              count        = DeltaCount,
              transient    = Transient,
              end_seq_id   = DeltaSeqIdEnd } = Delta,
+    %% We limit the number of messages we read from the index. We want a large
+    %% enough value but not one that will use too much memory because when there
+    %% are many queues we may end up exploding the memory. We can use the current
```

Comment:
How does a large enough value help?

When you write `exploding the memory`, do you mean that we may keep too many messages in memory unnecessarily, and not be memory frugal, as lazy queues are expected to be?

Reply:
It's about not reading every time a message must be accessed. Note that some operations will end up needing multiple messages at once (dropping expired messages, fetching and prefetching, for example). Before, those operations could end up reading 16k messages which were not necessarily used afterwards. Now we read potentially much less. The function is called again if more messages are needed.

Reply:
Got it 👍🏻